config/config.go

```diff
@@ -131,22 +131,23 @@ func SetVersion(ver string) {
 }
 
 // ReadNetclientConfig loads the host configuration into memory.
-func ReadNetclientConfig() {
+func ReadNetclientConfig() error {
 	lockfile := filepath.Join(os.TempDir(), ConfigLockfile)
 	file := GetNetclientPath() + "netclient.yml"
 	if err := Lock(lockfile); err != nil {
 		logger.Log(0, "unable to obtain lockfile for host config", err.Error())
-		return
+		return err
```

Comment:
if you are going to return an err, you should probably not log the err here and let the caller do that if required.